tests/Unit/TolerantVariableReferenceFinderTest.php

```diff
@@ -0,0 +1,201 @@
+            $uri,
+        ];
+
+        yield 'scope: inside class method' => [
+            '<?php class C1 { function M1(<sr>$var1) { <sr>$v<>ar1 = 5; $var2 = <sr>$var1 + 10; } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: select argument' => [
+            '<?php class C1 { function M1(<sr>$va<>r1) { <sr>$var1 = 5; $var2 = <sr>$var1 + 10; } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous function + use, click inside' => [
+            '<?php class C1 { function M1() { <sr>$var1 = 10; $f = function() use (<sr>$var1) { <sr>$v<>ar1 = 5; $var2 = <sr>$var1 + 10; } } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous function + use, click outside' => [
+            '<?php class C1 { function M1() { <sr>$v<>ar1 = 10; $f = function() use (<sr>$var1) { <sr>$var1 = 5; $var2 = <sr>$var1 + 10; } } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous function + use, click in use' => [
+            '<?php class C1 { function M1() { <sr>$var1 = 10; $f = function() use (<sr>$v<>ar1) { <sr>$var1 = 5; $var2 = <sr>$var1 + 10; } } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous function (no use), click inside' => [
+            '<?php class C1 { function M1() { $var1 = 10; $f = function(<sr>$var1) { <sr>$v<>ar1 = 5; $var2 = <sr>$var1 + 10; } } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous function (no use), click outside' => [
+            '<?php class C1 { function M1() { <sr>$v<>ar1 = 10; $f = function($var1) { $var1 = 5; $var2 = $var1 + 10; } } }',
+            $uri,
+        ];
+
+        yield 'scope: inside class method: inside anonumous class method' => [
+            '<?php '.
+                'class C1 { function M1() { '.
+                '$var = 1;'.
+                '$c = new class { function IM() { <sr>$v<>ar = 1; } } '.
+                ' } }',
+            $uri,
+        ];
+    }
+
+    private static function offsetsFromSource(string $source, ?string $uri): array
+    {
+        $textDocumentUri = $uri !== null ? TextDocumentUri::fromString($uri) : null;
+        $results = preg_split("/(<>|<sr>|<mr>)/u", $source, null, PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_NO_EMPTY);
+        
+        $referenceLocations = [];
+        $selectionOffset = null;
+
+        if (is_array($results)) {
+            $newSource = "";
+            $offset = 0;
+            foreach ($results as $result) {
+                if ($result == "<>") {
+                    $selectionOffset = $offset;
+                } elseif ($result == "<sr>") {
+                    $referenceLocations[] = PotentialLocation::surely(
+                        new Location($textDocumentUri, ByteOffset::fromInt($offset))
+                    );
+                } elseif ($result == "<mr>") {
```

Comment:
I don't think this finder needs "maybe"? Maybe would apply if we have a method `$foo->bar()` but we don't know the type of `$foo` so if we search for `Foo->bar()` it's possible for `Foo`, but it could be an identically named method.

I think in this case we can always be 100% sure that the variable is what we're looking for?

Reply:
Just thought that code can be generalised in the future, so I included it. I never use it.

Reply:
better to do as little as possible